cogs/fun.py

```diff
@@ -1,16 +1,288 @@
 import random
 import textwrap
-
+import urllib
+import typing as t
+import aiohttp
+import os
 import discord
-from discord import Color, Embed
+from random import choice, randint
+from discord import Color, Embed, Message
 from discord.ext.commands import BadArgument, Bot, BucketType, Cog, Context, command, cooldown
 
+from cogs.utils.errors import ServiceError
+
+from .endpoints.endpoints import nekos
+
+
+file = open("assets" + os.path.sep + "excuses.txt", "r", encoding="utf-8")
+excuses = file.readlines()
+file.close()
+
 
 class Fun(Cog):
     """This is a cog for simple fun commands."""
 
     def __init__(self, bot: Bot) -> None:
         self.bot = bot
+        self.user_agent = {"User-Agent": "HotWired"}
+        self.dadjoke = {
+            "User-Agent": "HotWired",
+            "Accept": "text/plain",
+        }
+
+    @command()
+    async def catfact(self, ctx: Context) -> None:
+        """Sends a random cat fact"""
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://cat-fact.herokuapp.com/facts") as response:
+                self.all_facts = await response.json()
+        fact = choice(self.all_facts["all"])
+        await ctx.send(embed=Embed(title="Did you Know?", description=fact["text"], color=0x690E8))
+
+    @command()
+    async def chuck(self, ctx: Context) -> t.Union[None, Message]:
+        """Get a random Chuck Norris joke"""
+        if randint(0, 1):
+            async with aiohttp.ClientSession() as session:
+                async with session.get("https://api.chucknorris.io/jokes/random") as r:
+                    joke = await r.json()
+                    return await ctx.send(joke["value"])
+        if ctx.guild:
+            if not ctx.channel.is_nsfw():
+                async with aiohttp.ClientSession() as session:
+                    async with session.get("http://api.icndb.com/jokes/random?exclude=[explicit]") as r:
+                        joke = await r.json()
+                        return await ctx.send(joke["value"]["joke"])
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get("http://api.icndb.com/jokes/random") as r:
+                joke = await r.json()
+                await ctx.send(joke["value"]["joke"].replace("&quote", '"'))
+
+    @command()
+    async def dong(self, ctx: Context, dick: discord.Member = None) -> None:
+        """How long is this person's dong ?"""
+        if not dick:
+            dick = ctx.author
+        await ctx.send(f"{dick.mention}'s magnum dong is this long : 8{'=' * randint(0, 15)}>")
+
```

Comment:
```suggestion
```
Not a big fan of this command,

Reply:
Sure :) Won't argue anymore, just will work